src/QAT/qat/purr/integrations/tket.py

```diff
@@ -380,27 +403,19 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
             architecture = Architecture([val.direction for val in couplings])
             optimizations_failed = not optimize_circuit(circ, architecture, opts)
         else:
-            min_quality = min([val.quality for val in couplings], default=0)
-            max_quality = max([val.quality for val in couplings], default=0)
-            chunks = (max_quality - min_quality) / 3
-            steps = 2
-            last_couplings = 0
-            while steps >= 0:
-                quality_level = min_quality + chunks * steps
-                filtered_couplings = [
-                    val.direction for val in couplings if val.quality >= quality_level
-                ]
-                steps = steps - 1
-                if not any(filtered_couplings) or last_couplings == len(
-                    filtered_couplings
-                ):
-                    continue
-
-                # Attempt optimization with good quality couplings, then degrade every
-                # failed try until we use everything.
-                last_couplings = len(filtered_couplings)
-                architecture = Architecture(filtered_couplings)
-                optimizations_failed = not optimize_circuit(circ, architecture, opts)
+            coupling_qualities = list({val.quality for val in couplings})
+            coupling_qualities.sort(reverse=True)
+            for quality_level in coupling_qualities:
```

Comment:
scenario when qualities are pairwise different: First iteration will consider **only one** coupling and subsequent iterations will include one more coupling at a time until optimisation succeeds. While this is correct, the filtering in this setup is at its finest granularity and I think `optimizations_failed` will remain `True` until you include enough couplings to actually make sth happen on the circuit.

Reply:
> scenario when qualities are pairwise different: First iteration will consider **only one** coupling and subsequent iterations will include one more coupling at a time until optimisation succeeds. While this is correct, the filtering in this setup is at its finest granularity and I think `optimizations_failed` will remain `True` until you include enough couplings to actually make sth happen on the circuit.

Previous code did the batching in 3 levels with which tket opt had a higher chance to "converge" quickly

Reply:
I don't understand what the question/point is.

Reply:
yeah basically increasing the batch size would help a lot